components/scream/src/physics/rrtmgp/rrtmgp_inputs_initializer.cpp

```diff
@@ -0,0 +1,189 @@
+
+        // These are actually outputs, but we need to initialize them in the ad
+        count += m_fields.count("sw_flux_up");
+        count += m_fields.count("sw_flux_dn");
+        count += m_fields.count("sw_flux_dn_dir");
+        count += m_fields.count("lw_flux_up");
+        count += m_fields.count("lw_flux_dn");
+  
+        if (count==0) {
+          return;
+        }
+  
+        EKAT_REQUIRE_MSG(
+            count==18,
+            "Error! RRTMGPInputsInitializer is expected to init\n"
+            "       pmid, pint, tmid, tint, col_dry, gas_vmr, sfc_alb_dir, sfc_alb_dif,\n"
+            "       mu0, lwp, iwp, rel, rei,\n"
+            "       sw_flux_up, sw_flux_dn, sw_flux_dn_dir, lw_flux_up, lw_flux_dn,\n"
+            "       but only " + std::to_string(count) + " of those have been found.\n"
+            "       Please, check the atmosphere processes you are using,\n"
+            "       and make sure they agree on who's initializing each field.\n"
+        );
+  
+        // Get device views
+        auto d_pmid = m_fields.at("pmid").get_view();
+        auto d_pint = m_fields.at("pint").get_view();
+        auto d_tmid = m_fields.at("tmid").get_view();
+        auto d_tint = m_fields.at("tint").get_view();
+        auto d_col_dry = m_fields.at("col_dry").get_view();
+        auto d_gas_vmr = m_fields.at("gas_vmr").get_view();
+        auto d_sfc_alb_dir = m_fields.at("sfc_alb_dir").get_view();
+        auto d_sfc_alb_dif = m_fields.at("sfc_alb_dif").get_view();
+        auto d_mu0 = m_fields.at("mu0").get_view();
+        auto d_lwp = m_fields.at("lwp").get_view();
+        auto d_iwp = m_fields.at("iwp").get_view();
+        auto d_rel = m_fields.at("rel").get_view();
+        auto d_rei = m_fields.at("rei").get_view();
+        auto d_sw_flux_up = m_fields.at("sw_flux_up").get_view();
+        auto d_sw_flux_dn = m_fields.at("sw_flux_dn").get_view();
+        auto d_sw_flux_dn_dir = m_fields.at("sw_flux_dn_dir").get_view();
+        auto d_lw_flux_up = m_fields.at("lw_flux_up").get_view();
+        auto d_lw_flux_dn = m_fields.at("lw_flux_dn").get_view();
+  
+        // Copies of fields for the call to the RRTMGP input initializer function, since it redefines the arrays
+        real2d p_lay;
+        real2d t_lay;
+        real2d p_lev;
+        real2d t_lev;
+        real2d col_dry;
+        real2d sfc_alb_dir;
+        real2d sfc_alb_dif;
+        real1d mu0;
+        real2d lwp;
+        real2d iwp;
+        real2d rel;
+        real2d rei;
+
+        // NOTE: set ncol to size of col_flx dimension in the input file. This is so
+        // that we can compare to the reference data provided in that file. Note that
+        // this will copy the first column of the input data (the first profile) ncol
+        // times. We will then fill some fraction of these columns with clouds for
+        // the test problem.
+        // TODO: get this from the input data, or from the yaml file
```

Comment:
I do this in the p3 inputs initializer here: https://github.com/E3SM-Project/scream/pull/804/files#diff-2358bb740d72ad4f39acb43909312d0599b402efd314715660dfd482ab348e62R230